File: Data Structures & Algorithms/check-if-array-is-sorted-and-rotated/submission-0.cpp
class Solution {
   public:
    bool check(vector<int>& nums) {
        int count = 0;

        for (int i = 0; i < 2 * nums.size(); i++) {
            if (nums[(i - 1) % nums.size()] <= nums[i % nums.size()]) {
                count += 1;
            } else {
                count = 1;
            }

            if (count == nums.size()) return true;
        }

        return nums.size() == 1;
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        n = len(nums)
        result = []

        def dfs(i, cur, total):
            if total > target or i >= n:
                return

            if total == target:
                result.append(cur.copy())
                return
            
            cur.append(nums[i])
            dfs(i, cur, total+nums[i])
            cur.pop()
            dfs(i+1, cur, total)
            
        
        dfs(0,[],0)
        return result
        